### PR TITLE
Add HTK -> MRTK API porting guide

### PR DESCRIPTION
**Pull request by keveleigh**
Overview
---
Adds an API / feature-level porting guide from HTK -> MRTK.

Fixes: #3437

### CODE REVIEW

File: Documentation/GettingStartedWithTheMRTK.md [hunk 1]
```diff
@@ -17,7 +17,7 @@ To get started with the Mixed Reality Toolkit you will need:
 
 There is not a direct upgrade path from the HoloToolkit to the new Mixed Reality Toolkit due to the rebuilt framework.  However, it is possible to import the MRTK into your HoloToolkit project and start work migrating your implementation if you wish.
 
-> The Mixed Reality Team will release a guide in the future on the key differences between HTK and MRTK project implementations.
+* [HTK to MRTK Porting Guide](HTKToMRTKPortingGuide.md)
```
Comment (davidkline-ms) [inline Documentation/GettingStartedWithTheMRTK.md]:
```suggestion
* [HoloToolkit to Mixed Reality Toolkit Porting Guide](HTKToMRTKPortingGuide.md)
```

File: Documentation/HTKToMRTKPortingGuide.md [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+| `ISourcePositionHandler` | [`IMixedRealityInputHandler<Vector3>`](xref:Microsoft.MixedReality.Toolkit.Input.IMixedRealityInputHandler`1) or [`IMixedRealityInputHandler<MixedRealityPose>`](xref:Microsoft.MixedReality.Toolkit.Input.IMixedRealityInputHandler`1) | Mapped to pointer position or grip position |
+| `ISourceRotationHandler` | [`IMixedRealityInputHandler<Quaternion>`](xref:Microsoft.MixedReality.Toolkit.Input.IMixedRealityInputHandler`1) or [`IMixedRealityInputHandler<MixedRealityPose>`](xref:Microsoft.MixedReality.Toolkit.Input.IMixedRealityInputHandler`1) | Mapped to pointer position or grip position |
+| `ISourceStateHandler` | [`IMixedRealitySourceStateHandler`](xref:Microsoft.MixedReality.Toolkit.Input.IMixedRealitySourceStateHandler) | |
+| `IXboxControllerHandler` | [`IMixedRealityInputHandler`](xref:Microsoft.MixedReality.Toolkit.Input.IMixedRealityInputHandler) and [`IMixedRealityInputHandler<Vector2>`](xref:Microsoft.MixedReality.Toolkit.Input.IMixedRealityInputHandler`1) | Mapped to the various controller buttons and thumbsticks |
+
+## Camera
```
Comment (davidkline-ms) [inline Documentation/HTKToMRTKPortingGuide.md]: should have a related profiles list

File: Documentation/HTKToMRTKPortingGuide.md [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
+|                           | HTK 2017 |  MRTK v2  |
+|---------------------------|----------|-----------|
+| Setup                     | Delete MainCamera, add MixedRealityCameraParent / MixedRealityCamera / HoloLensCamera prefab to scene **or** use Mixed Reality Toolkit > Configure > Apply Mixed Reality Scene Settings menu item. | MainCamera parented under MixedRealityPlayspace via Mixed Reality Toolkit > Add to Scene and Configure... |
+| Configuration             | Camera settings configuration performed on prefab instance. | Camera settings configured in the [Mixed Reality Camera Profile](xref:Microsoft.MixedReality.Toolkit.MixedRealityCameraProfile). |
+
+## Speech
```
Comment (davidkline-ms) [inline Documentation/HTKToMRTKPortingGuide.md]: needs related profiles listed

File: Documentation/HTKToMRTKPortingGuide.md [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+| Event handlers            | `ISpeechHandler` | [`IMixedRealitySpeechHandler`](xref:Microsoft.MixedReality.Toolkit.Input.IMixedRealitySpeechHandler) |
+
+### Dictation
+
+|                           | HTK 2017 |  MRTK v2  |
+|---------------------------|----------|-----------|
+| Setup                     | Add a DictationInputManager to your scene. | Dictation support requires service (e.g., Windows Dictation Input Manager) to be added to the Input System's data providers. |
+| Event handlers            | `IDictationHandler` | `IMixedRealityDictationHandler`[`IMixedRealitySpeechHandler`](xref:Microsoft.MixedReality.Toolkit.Input.IMixedRealitySpeechHandler) |
+
+## Spatial Awareness / Mapping
```
Comment (davidkline-ms) [inline Documentation/HTKToMRTKPortingGuide.md]: Related profiles needed

File: Documentation/HTKToMRTKPortingGuide.md [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+
+### Mesh
```
Comment (davidkline-ms) [inline Documentation/HTKToMRTKPortingGuide.md]: list mesh observer profile?

File: Documentation/HTKToMRTKPortingGuide.md [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+| Setup                     | Use the `SurfaceMeshesToPlanes` script. | Not yet implemented. |
+
+### Spatial Understanding
+
+|                           | HTK 2017 |  MRTK v2  |
+|---------------------------|----------|-----------|
+| Setup                     | Add the SpatialUnderstanding prefab to the scene. | Not yet implemented. |
+| Configuration             | Configure the scene instance in the inspector. | Not yet implemented. |
+
+## Boundary
```
Comment (davidkline-ms) [inline Documentation/HTKToMRTKPortingGuide.md]: profile(s) list

**Comment by cre8ivepark** [comment]
Added UX building blocks table. Thanks!

**Comment by wiwei** [comment]
Pulling this in so that we have the initial docs for this, we can follow up with improvements on in ongoing basis.

**Pull request merged**